src/protocol/capi/ConnectProtocol.cpp

```diff
@@ -428,6 +428,7 @@ namespace capi
 
       compressionHandler(options);
       setConnectionAttributes(options->connectionAttributes);
+      mysql_optionsv(connection.get(), MYSQL_INIT_COMMAND, (void*)options->initCommand.c_str());
```

Comment:
This is probably not the correct place to apply the option. I don't know where else it would fit better.